metadata/metadata.go

```diff
@@ -23,31 +31,66 @@ var (
 )
 
 func InitMetadata(ctx context.Context) error {
-	initK8sRKESystemImages()
-	initAddonTemplates()
-	initServiceOptions()
-	initDockerOptions()
+	data, err := loadData()
+	if err != nil {
+		return err
+	}
+	initK8sRKESystemImages(data)
+	initAddonTemplates(data)
+	initServiceOptions(data)
+	initDockerOptions(data)
 	return nil
 }
 
+// this method loads metadata from three place, if METADATA_URL is provided then load data from specified location. Otherwise load data from bindata.
+func loadData() (kdm.Data, error) {
+	var b []byte
+	var err error
+	u := os.Getenv(metadataURLEnv)
+	if u != "" {
+		b, err = readFile(u)
+		if err != nil {
+			return kdm.Data{}, err
```

Comment:
Can we describe this error better, its possible this error does not describe what is actually failing (just a HTTP response but not describing what it is trying to do)

Reply:
Update with detailed error message for func `loadData`.